File: Preprocessing.Cs/Preprocessing/Graph.cs
namespace AstGenerator;

public record Graph(string FilePath, string Label, string MethodName, int LineNumber, int Depth,
    List<string> OriginalCode, List<NodeData> Nodes,
    List<EdgeData> AstEdges, List<EdgeData> CfgEdges, List<EdgeData> CdgEdges, List<EdgeData> ReachingDefEdges);

File: Preprocessing.Java/src/main/java/org/example/Graph.java
package org.example;

import java.util.List;

public record Graph(String filePath, String label, String methodName, int lineNumber, int depth,
                    List<String> originalCode, List<NodeData> nodes,
                    List<EdgeData> astEdges, List<EdgeData> cfgEdges, List<EdgeData> cdgEdges,
                    List<EdgeData> reachingDefEdges) {
}
